robot/framesystem/framesystem.go

```diff
@@ -31,16 +31,48 @@ var API = resource.APINamespaceRDKInternal.WithServiceType(SubtypeName)
 var InternalServiceName = resource.NewName(API, "builtin")
 
 // A Service that returns the frame system for a robot.
+//
+// TransformPose example:
+//
+//	// Define a Pose coincident with the world reference frame
+//	firstPose := spatialmath.NewPoseFromPoint(r3.Vector{X: 0.0, Y: 0.0, Z: 0.0})
+//
+//	// Establish the world as the reference for firstPose
+//	firstPoseInFrame := referenceframe.NewPoseInFrame(referenceframe.World, firstPose)
+//
+//	// Calculate firstPoseInFrame from the perspective of the origin frame of myArm
+//	transformedPoseInFrame, err := machine.TransformPose(context.Background(), firstPoseInFrame, "myArm", nil)
```

Comment:
[nit]: this should be 
```
transformedPoseInFrame, err := fsService.TransformPose(context.Background(), firstPoseInFrame, "myArm", nil)
```

Reply:
I am being pedantic here because we don't use `machine` which I would assume means robot.

Reply:
hmmmmmmmmm, I might take it back I'm realized that we just do:
```
	robot, err := client.New(
		context.Background(),
		"sensor-controlled-base-main.2d8xnaiovq.viam.cloud",
		logger,
		client.WithDialOptions(rpc.WithEntityCredentials(
			"af540b1f-70c7-4119-9f49-95ab45e4ca10",
			rpc.Credentials{
				Type:    rpc.CredentialsTypeAPIKey,
				Payload: "vycm94zsiso49k8v986f0ya0xr81gvbu",
			})),
	)
	if err != nil {
		logger.Fatal(err)
	}
	robot.TransformPose()....
```

I take back my comment, what you have is easier to understand given a user looks at our code snippet on app.viam

Reply:
hmmmmmmmmm, I've realized that we just do:
```
	robot, err := client.New(
		context.Background(),
		"XXXX",
		logger,
		client.WithDialOptions(rpc.WithEntityCredentials(
			"XXXX",
			rpc.Credentials{
				Type:    rpc.CredentialsTypeAPIKey,
				Payload: "XXXX",
			})),
	)
	if err != nil {
		logger.Fatal(err)
	}
	robot.TransformPose()....
```

I take back my comment, what you have is easier to understand given a user looks at our code snippet on app.viam